pkg/policy/repository.go

```diff
@@ -71,17 +71,33 @@ func (p *Repository) GetPolicyCache() *PolicyCache {
 	return p.policyCache
 }
 
+type identityCacheProvider interface {
+	GetIdentityCache() cache.IdentityCache
+}
+
 // NewPolicyRepository allocates a new policy repository
-func NewPolicyRepository() *Repository {
+func NewPolicyRepository(idCacheProvider identityCacheProvider) *Repository {
```

Comment:
Would you be OK with me working on the interface plumbing in a follow-up PR? I can remove the existing one I have for now if you want for consistency.

Reply:
+1 for this! I was also wondering about it. Why are we passing `idCacheProvider identityCacheProvider` at all? In reality we should be passing a `selectorCache *SelectorCache`. Furthermore, what we really should pass is an interface that implements all methods of `SelectorCache`

Reply:
The reason I did it this way was to maintain the exact same behavior (e.g., we call `GetIdentityCache` at exactly the same place) to avoid any change in behavior apart from function signature. I'll see if we can pass the `IdentityCache` in directly and get the same results. @aanm the motivation behind passing in a `*SelectorCache` is unclear to me, can you elaborate on its benefits please? I am not opposed to doing it that way, but don't see it as a clear benefit over what we have now.

Reply:
@ianvernon it's weird to see `NewPolicyRepository` that has to create a SelectorCache based on the IdentityCache. The PolicyRepository's constructor should not create the SelectorCache because in reality it doesn't use it for anything, it is just used to return it when `GetPolicyCache()` (I believe this is the methods' name, I'm talking about the only method that returns a SelectorCache) is executed. That's why I said that this should even be returning an interface of the implementation of SelectorCache.

Reply:
I'll post a commit which changes `NewPolicyRepository` to take in an `IdentityCache` directly. 

Re: your comment around how the policy repository is created - I think it makes sense to initialize the policy repository type with any known state that affects its internal members at daemon bootstrap, specifically identities, as the identities that are currently present dictate the state of the `SelectorCache` member; a structure which is dependent upon state from other systems should be initialized with said state when the structure is created.

Re: your statement in the comment that the "policy repository does not use the selector cache for anything" - I think the the type name `Repository` has become a bit out of sync which it actually does - we have a type which manages the state of all rules (creation, deletion, caching of information about selectors within the rules, etc.). However, the state of one thing (the list of rules) directly influences the state of the `SelectorCache`, so coupling the two together conceptually makes sense to me.

I also think that the changes being proposed here are out of scope of what this PR is trying to achieve. This PR serves to reduce package-level imports / creation of more explicit types within Cilium, and does that.